server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -8,75 +8,90 @@
 
+import static org.elasticsearch.index.query.AbstractQueryBuilder.DEFAULT_BOOST;
+
 /**
- * A {@link FieldMapper} for indexing a sparse vector of floats.
- *
- * @deprecated The sparse_vector type was deprecated in 7.x and removed in 8.0. This mapper
- * definition only exists so that 7.x indices can be read without error.
- *
- * TODO: remove in 9.0.
+ * A {@link FieldMapper} that exposes Lucene's {@link FeatureField} as a sparse
+ * vector of features.
  */
-@Deprecated
 public class SparseVectorFieldMapper extends FieldMapper {
-    private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(SparseVectorFieldMapper.class);
-    static final String ERROR_MESSAGE = "The [sparse_vector] field type is no longer supported.";
-    static final String ERROR_MESSAGE_7X = "The [sparse_vector] field type is no longer supported. Old 7.x indices are allowed to "
-        + "contain [sparse_vector] fields, but they cannot be indexed or searched.";
+
     public static final String CONTENT_TYPE = "sparse_vector";
 
+    static final String ERROR_MESSAGE_7X = "[sparse_vector] field type in old 7.x indices is allowed to "
+        + "contain [sparse_vector] fields, but they cannot be indexed or searched.";
+
+    private static SparseVectorFieldType ft(FieldMapper in) {
+        return ((SparseVectorFieldMapper) in).fieldType();
+    }
+
     public static class Builder extends FieldMapper.Builder {
 
-        final Parameter<Map<String, String>> meta = Parameter.metaParam();
+        private final Parameter<Boolean> positiveScoreImpact = Parameter.boolParam(
+            "positive_score_impact",
+            false,
+            m -> ft(m).positiveScoreImpact,
+            true
+        );
+        private final Parameter<Map<String, String>> meta = Parameter.metaParam();
 
         public Builder(String name) {
             super(name);
         }
 
         @Override
         protected Parameter<?>[] getParameters() {
-            return new Parameter<?>[] { meta };
+            return new Parameter<?>[] { positiveScoreImpact, meta };
         }
 
         @Override
         public SparseVectorFieldMapper build(MapperBuilderContext context) {
             return new SparseVectorFieldMapper(
                 name,
-                new SparseVectorFieldType(context.buildFullName(name), meta.getValue()),
+                new SparseVectorFieldType(context.buildFullName(name), meta.getValue(), positiveScoreImpact.getValue()),
                 multiFieldsBuilder.build(this, context),
-                copyTo.build()
+                copyTo.build(),
+                positiveScoreImpact.getValue()
             );
         }
     }
 
     public static final TypeParser PARSER = new TypeParser((n, c) -> {
-        if (c.indexVersionCreated().onOrAfter(IndexVersion.V_8_0_0)) {
-            throw new IllegalArgumentException(ERROR_MESSAGE);
-        } else {
+        if (c.indexVersionCreated().before(IndexVersion.V_8_0_0)) {
```

Comment:
I added checks that were present on indices created before 8.0

Reply:
we should throw for anything between `[8.0.0..latest)` as during those versions it was not allowed.

Reply:
Added that, thanks!